validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/LocalValidatorSource.java

```diff
@@ -83,25 +88,37 @@ public LocalValidatorSource(
   }
 
   @Override
-  public boolean canAddValidator() {
+  public boolean canUpdateValidators() {
     return !readOnly && maybeDataDirLayout.isPresent();
   }
 
+  @Override
+  public DeleteKeyResult deleteValidator(final BLSPublicKey publicKey) {
+    if (!canUpdateValidators()) {
+      return DeleteKeyResult.error(
+          "Cannot delete validator from read-only local validator source.");
+    }
+    final Optional<ActiveLocalValidatorSource> source =
+        Optional.ofNullable(localValidatorSourceMap.remove(publicKey));
+    if (source.isEmpty()) {
+      return DeleteKeyResult.error(
+          "Could not find " + publicKey.toBytesCompressed().toShortHexString() + " to delete");
+    }
+    keystoreLocker.unlockKeystore(getKeystorePath(publicKey));
```

Comment:
I think we should only unlock if the delete succeeded.  It should matter as we should have deleted from memory before now anyway, but the safe option is to leave the lock in place incorrectly whereas deleting it incorrectly could be dangerous.

Reply:
it's out of memory so not locked at that point, but can make that change.

Reply:
Yeah agreed, I think it would be fine either way. But it's probably polite to have the lock on the file when we're deleting it rather than potentially deleting it out from under someone else.